In [7]:
import pandas as pd

df = pd.read_csv("train.tsv", delimiter='\t', header=None, names=['text', 'label'])
print(df.shape)
print(df.head())

(6920, 2)
                                                text  label
0  a stirring , funny and finally transporting re...      1
1  apparently reassembled from the cutting room f...      0
2  they presume their audience wo n't sit still f...      0
3  this is a visually stunning rumination on love...      1
4  jonathan parker 's bartleby should have been t...      1


In [8]:
print("Total reviews:", df.shape[0])
print("Columns:", df.shape[1])
print("\nLabel distribution:")
print(df['label'].value_counts())
print("\nSample review:")
print(df['text'][0])


Total reviews: 6920
Columns: 2

Label distribution:
label
1    3610
0    3310
Name: count, dtype: int64

Sample review:
a stirring , funny and finally transporting re imagining of beauty and the beast and 1930s horror films


In [9]:
df['length'] = df['text'].apply(lambda x: len(x.split()))

print("Avg words per review:", round(df['length'].mean()))
print("Shortest review:", df['length'].min(), "words")
print("Longest review:", df['length'].max(), "words")


Avg words per review: 19
Shortest review: 1 words
Longest review: 51 words


# Day 1 — Environment Setup & Data Loading

**Dataset:** SST2 movie reviews (6920 samples)  
**Labels:** 1 = positive (3610), 0 = negative (3310)  
**Avg review length:** 19 words  

Data is balanced and clean. Ready for EDA.